src/Polytopes/LinearProgram.jl

```diff
@@ -0,0 +1,147 @@
+@doc Markdown.doc"""
+    LinearProgram(P, c [, k])
+
+The linear program on the feasible set P (a Polyhedron) with
+ respect to the function x ↦ dot(c,x)+k where k is optional (default 0).
+
+""" struct LinearProgram
+   feasible_region::Polyhedron
+   polymake_lp::Polymake.BigObjectAllocated
+   convention::Symbol
+   function LinearProgram(Q::Polyhedron, objective::AbstractVector, k; convention = :max)
+      P=Polyhedron(Polymake.polytope.Polytope(pm_polytope(Q)))
+      ambDim = ambient_dim(P)
+      size(objective, 1) == ambDim || error("objective has wrong dimension.")
+      lp = Polymake.polytope.LinearProgram(LINEAR_OBJECTIVE=homogenize(objective, k))
+      pm_polytope(P).LP = lp
+      new(P, lp, convention)
+   end
+   function LinearProgram(Q::Polyhedron, objective::AbstractVector; convention = :max)
+      LinearProgram(Q,objective,0; convention=convention)
+   end
+end
+
+
+###############################################################################
+###############################################################################
+### Display
+###############################################################################
+###############################################################################
+function Base.show(io::IO, lp::LinearProgram)
+    c=dehomogenize(lp.polymake_lp.LINEAR_OBJECTIVE)
+    k=lp.polymake_lp.LINEAR_OBJECTIVE[1]
+    print(io, "The linear program\n")
+    if lp.convention == :max
+      print(io, "   max")
+    elseif lp.convention == :min
+      print(io, "   min")
+    end
+    print(io, "{dot(c,x)+k | x ∈ P}\n")
+    print(io, "where P is a "*string(typeof(lp.feasible_region)))
+    print(io, " and\n   c=")
+    print(io, string(c'))
+    print(io, "\n   k=")
+    print(io, string(k))
+end
+
+
+###############################################################################
+###############################################################################
+### Access
+###############################################################################
+###############################################################################
+
+"""
+    objective_function(LP::LinearProgram; as = :pair)
+
+Returns the objective function x ↦ dot(c,x)+k of the linear program LP.
+The allowed values for `as` are
+* `pair`: Returns the pair `(c,k)`
+* `function`: Returns the objective function as a function.
+
+
+"""
+function objective_function(LP::LinearProgram; as = :pair)
+   if as == :pair
+      return(dehomogenize(LP.polymake_lp.LINEAR_OBJECTIVE),LP.polymake_lp.LINEAR_OBJECTIVE[1])
+   elseif as == :function
+      (c,k) = objective_function(LP, as = :pair)
+      function f(x)
+         eval=sum(x.*c)+k
+      end
+      return(f)
```

Comment:
more idiomatic might be this:
```suggestion
      return x -> sum(x.*c)+k
```